apis/vmware/v1beta1/vspheremachinenetworkstatus_types.go

```diff
@@ -0,0 +1,216 @@
+	//
+	// +optional
+	Lifetime metav1.Time `json:"lifetime,omitempty"`
+
+	// Origin describes how this address was configured.
+	//
+	// +optional
+	// +kubebuilder:validation:Enum=dhcp;linklayer;manual;other;random
+	Origin string `json:"origin,omitempty"`
+
+	// State describes the state of this IP address.
+	//
+	// +optional
+	// +kubebuilder:validation:Enum=deprecated;duplicate;inaccessible;invalid;preferred;tentative;unknown
+	State string `json:"state,omitempty"`
+}
+
+// VSphereMachineNetworkInterfaceIPStatus describes the observed state of a
+// VM's network interface's IP configuration.
+type VSphereMachineNetworkInterfaceIPStatus struct {
+	// AutoConfigurationEnabled describes whether or not ICMPv6 router
+	// solicitation requests are enabled or disabled from a given interface.
+	//
+	// These requests acquire an IP6 address and default gateway route from
+	// zero-to-many routers on the connected network.
+	//
+	// If not set then ICMPv6 is not available on this VM.
+	//
+	// +optional
+	AutoConfigurationEnabled *bool `json:"autoConfigurationEnabled,omitempty"`
+
+	// DHCP describes the VM's observed, client-side, interface-specific DHCP
+	// options.
+	//
+	// +optional
+	DHCP *VSphereMachineNetworkDHCPStatus `json:"dhcp,omitempty"`
+
+	// Addresses describes observed IP addresses for this interface.
+	//
+	// +optional
+	Addresses []VSphereMachineNetworkInterfaceIPAddrStatus `json:"addresses,omitempty"`
+
+	// MACAddr describes the observed MAC address for this interface.
+	//
+	// +optional
+	MACAddr string `json:"macAddr,omitempty"`
+}
+
+// VSphereMachineNetworkInterfaceStatus describes the observed state of a
+// VM's network interface.
+type VSphereMachineNetworkInterfaceStatus struct {
+	// Name describes the corresponding network interface with the same name
+	// in the VM's desired network interface list. If unset, then there is no
+	// corresponding entry for this interface.
+	//
+	// Please note this name is not necessarily related to the name of the
+	// device as it is surfaced inside of the guest.
+	//
+	// +optional
+	Name string `json:"name,omitempty"`
+
+	// DeviceKey describes the unique hardware device key of this network
+	// interface.
+	//
+	// +optional
+	DeviceKey int32 `json:"deviceKey,omitempty"`
+
+	// IP describes the observed state of the interface's IP configuration.
+	//
+	// +optional
+	IP *VSphereMachineNetworkInterfaceIPStatus `json:"ip,omitempty"`
+
+	// DNS describes the observed state of the interface's DNS configuration.
+	//
+	// +optional
+	DNS *VSphereMachineNetworkDNSStatus `json:"dns,omitempty"`
+}
+
+// VSphereMachineNetworkStatus defines the observed state of a VM's
+// network configuration.
+//
+// This a mirror of the v1alpha2 VirtualMachineNetworkStatus. See
+// https://github.com/vmware-tanzu/vm-operator/blob/main/api/v1alpha2/virtualmachine_network_types.go
+// for more information. When vm-operator v1alpha2 is updated, this type need to be synchronized.
```

Comment:
> When vm-operator v1alpha2 is updated, this type need to be synchronized

just fyi. This won't entirely be possible. We can make compatible changes, but not breaking ones (e.g. change types of fields)

If there are breaking changes we cannot bump our apiVersion frequently, also if eventually we might have to work against multiple vm-operator apiVersions at the same time. But for now it's fine, just wanted to share